WalletWasabi/WabiSabi/Crypto/CredentialIssuer.cs

```diff
@@ -128,7 +134,7 @@ public ICommitableCredentialsResponse PrepareResponse(CredentialsRequest registr
 			// then there is a problem somewhere because this should not be possible.
 			if (Balance + registrationRequest.Delta < 0)
 			{
-				throw new WabiSabiCryptoException(WabiSabiCryptoErrorCode.NegativeBalance);
+				throw new InvalidOperationException("Negative issuer balance");
```

Comment:
Yes, the user will get `Internal Server Error` - that's not good. He should get something specific that we will be able to debug according to his log, plus the server log. I would change it back `WabiSabiCryptoException(WabiSabiCryptoErrorCode.NegativeBalance)`, as the concept was figured out in the past. 

https://github.com/molnard/WalletWasabi/blob/015eac53f66e478b74d79365786093cbe1907058/WalletWasabi.Backend/Filters/ExceptionTranslateFilter.cs#L14

Reply:
What can be considered is to have additional logs on the backend side.

Reply:
At the meeting, we made an agreement, that this exception should not be delivered to the user as it is indicating an invalid `CredentialIssuer` state which is clearly backend related, moreover the details should be kept on the backend as it can be a crypto bug or something that could be exposed.